thewalrus/internal_modes/fock_density_matrices.py

```diff
@@ -152,6 +157,7 @@ def density_matrix_single_mode(cov, pattern, normalize=False, LO_overlap=None, c
         LO_overlap (array): overlap between internal modes and local oscillator
         cutoff (int): photon number cutoff. Should be odd. Even numbers will be rounded up to an odd number
         hbar (float): the value of hbar (default 2)
+        method (str): which method to use, "recursive" or "direct"
```

Comment:
The options are really "recursive" and anything else. For example in the tests you use "non-recursive". In case of typos, perhaps it's better to have if "recursive":... elif "non-recursive":... else: raise Error in the code? Or to have kwarg recursive=True or something, unless we are likely to have more methods in the future.